src/box/iproto.cc

```diff
@@ -3397,6 +3490,26 @@ iproto_do_cfg_f(struct cbus_call_msg *m)
 		iproto_thread_accept(iproto_thread, io, addr, addrlen, session);
 		break;
 	}
+	case IPROTO_CFG_DROP_CONNECTIONS: {
+		struct iproto_connection *con;
+		static const struct cmsg_hop cancel_route[1] =
+						{{ tx_process_cancel, NULL }};
+		rlist_foreach_entry(con, &iproto_thread->connections,
+				    in_connections) {
+			if (!con->is_in_replication)
+				iproto_connection_close(con);
+			con->drop = true;
+			iproto_thread->drop_count++;
+			if (con->state != IPROTO_CONNECTION_DESTROYED) {
+				cmsg_init(&con->cancel_msg, cancel_route);
+				cpipe_push(&iproto_thread->tx_pipe,
+					   &con->cancel_msg);
+			}
```

Comment:
No need to set the `drop` flag for already destroyed connections.

Reply:
We want to wait when all connections are freed on shutdown.